### DISCUSSION TITLE
[archived] "Upgrading" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#12016] [General]
# argo-workflows/upgrading/

https://argoproj.github.io/argo-workflows/upgrading/

**Comment by ShyunnY** [comment]
How should I perform a smooth upgrade? Is there a corresponding guide?

**Reply by agilgur5** [reply]
If there aren't any specific notes listed for the version you're looking to upgrade to, you can just upgrade to the newest version in a standard k8s rolling upgrade. You just want to make sure you're using the same version everywhere: Controller, Server, Executor (if you've explicitly specified it anywhere), CLI, etc.
